conan/tools/gnu/autotools.py

```diff
@@ -3,69 +3,50 @@
 
 from conan.tools import CONAN_TOOLCHAIN_ARGS_FILE
 from conan.tools._compilers import use_win_mingw
+from conan.tools.gnu.make import make_jobs_cmd_line_arg
+from conans.client.build import join_arguments
 from conans.util.files import load
 
 
 class Autotools(object):
 
     def __init__(self, conanfile):
-        """
-        FIXME: include_rpath_flags CONAN 2.0 to default True? Could break many packages in center
-        """
         self._conanfile = conanfile
-        self._win_bash = False
-        self._include_rpath_flags = False
-        self._os = conanfile.settings.get_safe("os")
-        self._os_version = conanfile.settings.get_safe("os.version")
-        self._os_sdk = conanfile.settings.get_safe("os.sdk")
-        self._os_subsystem = conanfile.settings.get_safe("os.subsystem")
-        self._arch = conanfile.settings.get_safe("arch")
-        self._build_type = conanfile.settings.get_safe("build_type")
-        self._compiler = conanfile.settings.get_safe("compiler")
-        self._compiler_version = conanfile.settings.get_safe("compiler.version")
-        self._build = None
-        self._host = None
-        self._target = None
-        if os.path.isfile(CONAN_TOOLCHAIN_ARGS_FILE):
-            args = json.loads(load(CONAN_TOOLCHAIN_ARGS_FILE))
-            self._build = args["build"] if "build" in args else None
-            self._host = args["host"] if "host" in args else None
-            self._target = args["target"] if "target" in args else None
+        self.environment_files = ["conanbuildenv", "conanautotoolstoolchain", "conanautotoolsdeps",
```

Comment:
To discuss: I removed the `AutoolsGen` because I think it is easier to be explicit but we should discuss if this is the best idea of "applying" the environments or we can think of a better one. I have seen something similar in the Bazel one.

Reply:
I agree is nicer to be explicit for Conan, so good for me to remove it. The only thing we should consider is if there is other way to improve the UX of the developer flow that does ``conan install`` + activate several env scripts (order matter? should I activate all of them?)